ci/version.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: Apache-2.0
+
+import subprocess
+import sys
+import os
+
+
+def set_version(current_version, new_version, dry_run=False):
+    """Tries to update the version from current_version to new_version."""
+    print(f"Modifying version from {current_version} to {new_version}...")
+    if dry_run:
+        print("  (dry run, won't actually make the change)")
+
+    # Read the template.
+    with open("version-diff-template", "r") as f:
+        template = f.read()
+
+    # Fill out the template.
+    patch = template.format(frm=current_version, to=new_version).encode("utf-8")
```

Comment:
Why not use something like [`cargo bump`](https://crates.io/crates/cargo-bump) instead?

Reply:
I wasn't aware of it's existence, but...

> `thread 'main' panicked at 'Workspaces are not supported yet.'`

Also, it still wouldn't bump the version information for the Python bindings, in documentation, etc.

Reply:
I also tried from the subdirectories; it still fails. Also [the issue tracking this](https://github.com/wraithan/cargo-bump/issues/10) has been stale for 3 1/2 years. In fact, all of cargo bump [has been stale for 3 years](https://github.com/wraithan/cargo-bump/commits/master). It's not exactly a standard component of cargo.

I'm perfectly fine with replacing the script with that semantic release plugin, in which case this PR can just be closed. Though to be honest it seems like a lot of complexity for what looks to be a regex replacement, which in itself seems more likely to break than a patchfile... Anyway, do you have time to help me hammer out the semantic release logic sometime, or at least to get me started with it? I remember being very confused about the logic on the main repo.